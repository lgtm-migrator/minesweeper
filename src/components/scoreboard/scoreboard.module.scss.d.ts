// autogenerated by typings-for-css-modules-loader.
// Please do not change this file!
declare namespace ScoreboardModuleScssNamespace {
  export interface IScoreboardModuleScss {
    wrapperScoreboard: string;
  }
}

declare const ScoreboardModuleScssModule: ScoreboardModuleScssNamespace.IScoreboardModuleScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: ScoreboardModuleScssNamespace.IScoreboardModuleScss;
};

export = ScoreboardModuleScssModule;
