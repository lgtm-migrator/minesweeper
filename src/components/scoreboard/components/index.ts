export * from "./components/counter/counter";
export * from "./components/level";
