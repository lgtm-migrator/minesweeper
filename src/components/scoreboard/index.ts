export { ScoreBoard, ScoreboardProps } from "./scoreboard";
