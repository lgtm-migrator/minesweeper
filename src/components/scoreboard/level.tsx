import React, { FunctionComponent } from "react";

import styles from "./level.module.scss";

export const Level: FunctionComponent<{ levelData: string[] }> = ({ levelData }) => (
  <select className={styles.selectLevel}>
    {levelData.map((item: string) => {
      return (
        <option className={styles.optionLevel} key={item}>
          {item}
        </option>
      );
    })}
  </select>
);
