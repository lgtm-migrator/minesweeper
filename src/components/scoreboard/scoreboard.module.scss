.wrapperScoreboard {
  display: flex;
  width: max-content;
  padding-bottom: 2vw;
  justify-content: space-between;
}
