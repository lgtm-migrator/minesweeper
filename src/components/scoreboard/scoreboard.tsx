import React, { ChangeEvent, FunctionComponent } from "react";

import { Counter, Level, ResetButton } from "./components";
import styles from "./scoreboard.module.scss";

export interface ScoreboardProps {
  /**
   * Timer
   */
  time: string;
  /**
   * Possible game scenarios
   */
  levels: string[];
  /**
   * Action handler when the GameReset button is clicked
   */
  onReset: () => void;
  /**
   * Bombs in the field
   */
  mines: string;
  /**
   * Action handler when select new lvl
   */
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
}

export const ScoreBoard: FunctionComponent<ScoreboardProps> = ({ time, mines, onReset, levels, onChange }) => {
  return (
    <div className={styles.wrapperScoreboard}>
      <Counter itemNumber={time} />
      <div>
        <Level onChange={onChange} levelData={levels} />
        <ResetButton onReset={onReset} />
      </div>
      <Counter itemNumber={mines} />
    </div>
  );
};
