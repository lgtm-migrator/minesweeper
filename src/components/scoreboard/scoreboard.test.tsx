import { ScoreBoard } from "@components/scoreboard/scoreboard";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";

describe("Scoreboard test cases", () => {
  const levels = ["beginner", "intermediate", "expert"];

  it("Scoreboard renders correctly", () => {
    const { asFragment } = render(
      <ScoreBoard time="000" levels={levels} onReset={() => null} mines="010" onChangeLevel={() => null} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("Scoreboard select level handler check", async () => {
    const onChange = jest.fn();

    render(<ScoreBoard time="000" levels={levels} onReset={() => null} mines="010" onChangeLevel={onChange} />);

    await userEvent.selectOptions(screen.getByRole("select-component"), "expert");

    expect(screen.getByRole("option", { name: "expert" })).toBeEnabled();

    await Promise.resolve();
    expect(onChange).toHaveBeenCalled();
  });
});
