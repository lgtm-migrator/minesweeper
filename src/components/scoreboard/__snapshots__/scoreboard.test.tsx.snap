// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard test cases Scoreboard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapperScoreboard"
  >
    <div
      class="parentCounter"
    >
      000
    </div>
    <div>
      <select
        class="selectLevel"
        role="select-component"
      >
        <option
          class="optionLevel"
        >
          beginner
        </option>
        <option
          class="optionLevel"
        >
          intermediate
        </option>
        <option
          class="optionLevel"
        >
          expert
        </option>
      </select>
      <button
        class="resetButton"
      >
        ðŸ™‚
      </button>
    </div>
    <div
      class="parentCounter"
    >
      010
    </div>
  </div>
</DocumentFragment>
`;
